internal/service/ssm/parameter_ephemeral.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package ssm
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+const (
+	ERNameParameter = "Ephemeral Resource Parameter"
+)
+
+// @EphemeralResource(aws_ssm_parameter, name="Parameter")
+func newEphemeralParameter(_ context.Context) (ephemeral.EphemeralResourceWithConfigure, error) {
+	return &ephemeralParameter{}, nil
+}
+
+type ephemeralParameter struct {
+	framework.EphemeralResourceWithConfigure
+}
+
+func (e *ephemeralParameter) Metadata(_ context.Context, _ ephemeral.MetadataRequest, response *ephemeral.MetadataResponse) {
+	response.TypeName = "aws_ssm_parameter"
+}
+
+func (e *ephemeralParameter) Schema(ctx context.Context, _ ephemeral.SchemaRequest, response *ephemeral.SchemaResponse) {
+	response.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			names.AttrARN: schema.StringAttribute{
+				Required: true,
+			},
```

Comment:
We can add a custom type for ARN validation

```suggestion
			names.AttrARN: schema.StringAttribute{
				CustomType: fwtypes.ARNType,
				Required:   true,
			},
```